Fix some misc test cleanup

brycekahle opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

Cleans up some files left behind in `/tmp` after test exit

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

brycekahle committed:
Fix some test cleanup

paulcacheux commented on pkg/config/remote/service/service_test.go:
```diff
@@ -108,6 +108,7 @@ func newTestService(t *testing.T, api *mockAPI, uptane *mockUptane, clock clock.
 
 	dir, err := os.MkdirTemp("", "testdbdir")
 	assert.NoError(t, err)
+	t.Cleanup(func() { os.RemoveAll(dir) })
```
I think you can use https://pkg.go.dev/testing#T.TempDir for this

brycekahle committed:
Use t.TempDir() instead

brycekahle commented:
`TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestServiceBackoffFailure2208823789\001\remote-config.db: The process cannot access the file because it is being used by another process.`

@DataDog/remote-config It seems the tests are failing now because something still has a handle to `remote-config.db` open. I don't see any sort of cleanup for the `Service` object on shutdown, including the `db *bbolt.DB` field.

ameske committed:
[RCM] Add a Stop method to Service to allow for clean-up in tests (#15877)

* [RCM] Add a Stop method to Service to allow for clean-up in tests

The bolt DB is designed to be persistent for the RC service, but we
don't want it persisting between test runs. A PR made to attempt to
clean up the DB file creating during tests was failing as there was no
way to delete the DB due to the fact the instantiated test Service
instance still had a handle to it. In order to allow the test code to
clean up the test bolt DB, we need a way to shut down the Service and
tell it to close the DB. The new Stop method leverages the
context.CancelFunc we create when starting the main goroutine to both
stop the main goroutine and then shutdown the bolt DB to allow test
cleanup to proceed.

* Use testing.Cleanup() to clean up test code

Less code is best code

brycekahle committed:
empty commit
